File: python/Brunel/_affiliation.py

__all__ = ["Affiliation"]


def _setState(state, val, default=None):
    try:
        result = state[val]
        if result:
            return result
        else:
            return default
    except:
        return default


class Affiliation:
    """Holds information about an Affiliation in the network"""
    def __init__(self, props=None, getHook=None):
        self._getHook = getHook

        self.state = {
            "name": None,
            "id": None,
            "sources": [],
            "notes": [],
        }

        self.setState(props)

    def __str__(self):
        return f"Affiliation({self.getName()})"

    def __repr__(self):
        return self.__str__()

    def getID(self):
        return self.state["id"]

    def getName(self):
        return self.state["name"]

    def setState(self, state):
        if not state:
            return

        self.state["name"] = _setState(state, "name")
        self.state["id"] = _setState(state, "id")
        self.state["sources"] = _setState(state, "sources", [])
        self.state["notes"] = _setState(state, "notes", [])

    def toDry(self):
        return self.state

    @staticmethod
    def unDry(value):
        return Affiliation(value)

    @staticmethod
    def load(data):
        return Affiliation(data)

File: src/version7/Affiliation.js

import Dry from "json-dry";

function setState(val, def=null){
  if (val){
    return val;
  } else {
    return def;
  }
}

class Affiliation {
  constructor(props){
    this.state = {
      name: null,
      id: null,
      sources: [],
      notes: {},
    };

    this.setState(props);

    this._getHook = null;
  }

  getID(){
    return this.state.id;
  }

  setState(state){
    if (state){
      this.state.name = setState(state.name);
      this.state.id = setState(state.id);
      this.state.sources = setState(state.sources, []);
      this.state.notes = setState(state.notes, {})
    }
  }

  _updateHooks(hook){
    this._getHook = hook;
  }

  toString(){
    return `Affiliation(${this.getName()})`;
  }

  getName(){
    return this.state.name;
  }

  toDry(){
    return {value: this.state};
  }
};

Affiliation.unDry = function(value){
  return new Affiliation(value);
}

Dry.registerClass("Affiliation", Affiliation);

export default Affiliation;
